Does it work with React 18?

This repo using react 18 https://github.com/DefectingCat/code-hike-test fails with:

```
Unhandled Runtime Error
TypeError: Cannot read properties of null (reading 'querySelector')

Call Stack
getPanelSnapshot
node_modules/@code-hike/mdx/dist/components.esm.mjs (2875:0)
eval
node_modules/@code-hike/mdx/dist/components.esm.mjs (2864:109)
basicStateReducer
node_modules/react-dom/cjs/react-dom.development.js (16376:0)
updateReducer
node_modules/react-dom/cjs/react-dom.development.js (16500:0)
updateState
node_modules/react-dom/cjs/react-dom.development.js (16836:0)
Object.useState
node_modules/react-dom/cjs/react-dom.development.js (17762:0)
Object.useState
node_modules/react/cjs/react.development.js (1617:0)
useSnapshots
node_modules/@code-hike/mdx/dist/components.esm.mjs (2849:55)
useTransition
node_modules/@code-hike/mdx/dist/components.esm.mjs (2442:41)
EditorTween
node_modules/@code-hike/mdx/dist/components.esm.mjs (2927:37)
```

It works after changing react 18 to 17.

So far I've got:

- If I install it with yarn v1.22.18: it works
- If I try to install it with npm v8.5.0: installation fails
```
While resolving: mrck.dev@undefined
Found: react@18.1.0
node_modules/react
  react@"^18.1.0" from the root project

Could not resolve dependency:
peer react@"^16.3.0 || ^17" from react-share@4.4.0
node_modules/react-share
  react-share@"^4.4.0" from the root project

Fix the upstream dependency conflict, or retry
this command with --force, or --legacy-peer-deps
to accept an incorrect (and potentially broken) dependency resolution.
```

<details>
  <summary>Explanation object</summary>
  
```json
{
  "code": "ERESOLVE",
  "current": {
    "name": "react",
    "version": "18.1.0",
    "whileInstalling": {
      "name": "mrck.dev",
      "path": "/workspace/mrck.dev"
    },
    "location": "node_modules/react",
    "isWorkspace": false,
    "dependents": [
      {
        "type": "prod",
        "name": "react",
        "spec": "^18.1.0",
        "from": {
          "location": "/workspace/mrck.dev"
        }
      }
    ]
  },
  "currentEdge": {
    "type": "prod",
    "name": "react",
    "spec": "^18.1.0",
    "from": {
      "location": "/workspace/mrck.dev"
    }
  },
  "edge": {
    "type": "peer",
    "name": "react",
    "spec": "^16.3.0 || ^17",
    "error": "INVALID",
    "from": {
      "name": "react-share",
      "version": "4.4.0",
      "whileInstalling": {
        "name": "mrck.dev",
        "path": "/workspace/mrck.dev"
      },
      "location": "node_modules/react-share",
      "isWorkspace": false,
      "dependents": [
        {
          "type": "prod",
          "name": "react-share",
          "spec": "^4.4.0",
          "from": {
            "location": "/workspace/mrck.dev"
          }
        }
      ]
    }
  },
  "strictPeerDeps": false,
  "force": false
}
```

</details>

- If I force the install with `npm i --legacy-peer-deps`: installation works, but http://localhost:3000/blog/posts/test-mdx/ fails with the `Cannot read properties of null (reading 'querySelector')` error (which probably is a problem with `React.useRef`)
- If I disable `reactStrictMode` in `next.config.js` the error is gone, and the page works.

Still investigating if the issue is Code Hike breaking strict mode or a problem with the app dependencies.

--- a/packages/mdx/next.config.js
+++ b/packages/mdx/next.config.js
@@ -1,4 +1,5 @@
 module.exports = {
+  reactStrictMode: true,
   redirects() {
     return [
       {

--- a/packages/mdx/src/mini-editor/editor-shift.tsx
+++ b/packages/mdx/src/mini-editor/editor-shift.tsx
@@ -793,21 +793,21 @@ function useSnapshots(
 
   useLayoutEffect(() => {
     if (!prevSnapshot) {
-      // debugger
+      const parent = ref.current!
       setState(s => ({
         ...s,
         prevSnapshot: {
-          ...getPanelSnapshot(ref.current!, prev),
-          ...getTabsSnapshot(ref.current!, prev),
+          ...getPanelSnapshot(parent, prev),
+          ...getTabsSnapshot(parent, prev),
         },
       }))
     } else if (!nextSnapshot) {
-      // debugger
+      const parent = ref.current!
       setState(s => ({
         ...s,
         nextSnapshot: {
-          ...getPanelSnapshot(ref.current!, next),
-          ...getTabsSnapshot(ref.current!, next),
+          ...getPanelSnapshot(parent, next),
+          ...getTabsSnapshot(parent, next),
         },
       }))
     }

--- a/packages/mdx/src/smooth-code/copy-button.tsx
+++ b/packages/mdx/src/smooth-code/copy-button.tsx
@@ -43,7 +43,7 @@ export function CopyButton({
         <path
           strokeLinecap="round"
           strokeLinejoin="round"
-          strokeWidth="2"
+          strokeWidth="1.6px"
           d="M8 16H6a2 2 0 01-2-2V6a2 2 0 012-2h8a2 2 0 012 2v2m-6 12h8a2 2 0 002-2v-8a2 2 0 00-2-2h-8a2 2 0 00-2 2v8a2 2 0 002 2z"
         />
       )}

